docs/ipdk-tofino.md

```diff
@@ -0,0 +1,115 @@
+IPDK on tofino
+===============
+
+# Overview
+This document describes how to build and run IPDK on Tofino.
+
+# Requirements
+- Docker (Ubuntu 20.04 base image)
+- Intel P4Studio
+- Git
+
+# Setup
+The build and execution of infrap4d uses docker.
+
+## Docker
+
+### Pull the Ubuntu 20.04 docker base image
+> docker pull ubuntu:20.04
+
+### Create container
+> docker run --privileged --cap-add=ALL -it --name infrap4d --network host -d ubuntu:20.04 /bin/bash
+
+## Git
+
+### Clone IPDK networking-recipe
+> git clone --recursive git@github.com:ipdk-io/networking-recipe.git infrap4d
+
+## Intel P4Studio
+Get the Intel P4Studio SDE version 9.11. The untarred directory is henceforth referred to as **sde**. If using an Intel Tofino Reference platform, please also download the BSP package.
+
+> tar xvf bf-sde-9.11.0-cpr.tgz
+
+# Build
+Build is done within the infrap4d docker container created above.
+
+> docker exec -it infrap4d bash
+
+### Install basic utilities
+```
+apt-get update
+apt-get install sudo git cmake autoconf gcc g++ libtool python3 python3-dev python3-distutils
+```
+
+### Build and install infrap4d dependencies
+```
+cd infrap4d/setup
+cmake . -DCMAKE_INSTALL_PREFIX=/usr/local/
+make
+ldconfig
+```
+
+### Build Intel P4Studio SDE
+The below steps are minimal. For a more detailed installation procedure, please refer to the Intel P4Studio SDE Installation Guide
+```
+cd sde/p4studio
```

Comment:
I take it that `ipdk.yaml` is a new profile.  Am I correct in assuming that it enables `tdi`?

Also, could we include instructions for specifying the install-prefix when building the Tofino SDE? I don't have a copy of the _P4Studio SDE Installation Guide_.

Reply:
The listed command is a one time hands-off command used to build the entire SDE. I was told this command also allows a --install-prefix option, but it seems unavailable.

The above command basically combines 4 commands,
p4studio dependencies install
p4studio packages extract
p4studio configure asic tdi ^thrift-driver
p4studio build

In the 3d command above, you can specify the --install-prefix option.

Reply:
All of this and other details are included in the Installation guide. I would rather not include all those details here again.

Reply:
My concern is that internal customers will have access to a copy of the SDE but won't have any of the documentation (which is packaged separately). They need to know how to do the basics, which includes being able to manage the install directory.